### PR TITLE
Add files

### PR DESCRIPTION
**Pull request by atreidas1**

### CODE REVIEW

File: items/resources.py [hunk 1]
```diff
@@ -84,6 +84,7 @@ def create_detail(self, object_list, bundle):
     return user.is_superuser
 
   def update_detail(self, object_list, bundle):
+    print "update"
```
Comment (ghost) [inline items/resources.py]: prints is wrong in the final code

File: items/resources.py [hunk 2]
```diff
@@ -132,6 +157,33 @@ class Meta:
...
+
+  def prepend_urls(self):
+    return [
+      url(r'^%s/upload/$' % self._meta.resource_name, self.wrap_view('upload'), name='api_file_upload'),
+    ]
+
+  def upload(self, request, **kwargs):
+    self.method_check(request, allowed=['post'])
+    current_user = request.user
+    if current_user.is_anonymous():
```
Comment (ghost) [inline items/resources.py]: it's covered by `self.is_authenticated(request)`

File: items/models.py [hunk 3]
```diff
@@ -32,3 +34,20 @@ class Meta:
...
+
+  class Meta:
+    db_table = 'todo_files'
+
+  def __str__(self):
+    return self.filename
+
+
+@receiver(pre_delete, sender=File)
+def mymodel_delete(sender, instance, **kwargs):
```
Comment (NikitaSedyx) [inline items/models.py]: `mymodel_delete` is strange name here for me

File: items/resources.py [hunk 4]
```diff
@@ -91,6 +91,27 @@ def delete_detail(self, object_list, bundle):
...
+
+  def read_detail(self, object_list, bundle):
+    return self.get_user(bundle) in bundle.obj.group.users.all()
+
+  def update_detail(self, object_list, bundle):
+    return self.get_user(bundle) in bundle.obj.group.users.all()
+
+  def create_detail(self, object_list, bundle):
+    user = self.get_user(bundle)
+    return (user in bundle.obj.group.users.all()) or (user==bundle.obj.group.creator)
```
Comment (NikitaSedyx) [inline items/resources.py]: creator is always in users I think

File: items/resources.py [hunk 5]
```diff
@@ -115,6 +137,11 @@ class Meta:
     authorization = GroupAuthorization()
     paginator_class = Paginator
     limit = 10
+    always_return_data = True
+    filtering = {
+      'title': ('startswith', ),
```
Comment (NikitaSedyx) [inline items/resources.py]: why only `startwith`?

File: items/resources.py [hunk 6]
```diff
@@ -132,6 +159,32 @@ class Meta:
...
+    authorization = FileAuthorization()
+
+  def prepend_urls(self):
+    return [
+      url(r'^%s/upload/$' % self._meta.resource_name, self.wrap_view('upload'), name='api_file_upload'),
+    ]
+
+  def upload(self, request, **kwargs):
+    self.method_check(request, allowed=['post'])
+    if self.is_authenticated(request):
```
Comment (NikitaSedyx) [inline items/resources.py]: does it work? I think it will throw `Unauthorized` for authenticated users

File: todo_dj/urls.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
 from django.conf.urls import patterns, include, url
 from django.contrib import admin
 from tastypie.api import Api
-from items.resources import ItemResource, UserResource, LoginResource, RegistrationResource
+from items.resources import ItemResource, UserResource, LoginResource, RegistrationResource,  FileResource
```
Comment (NikitaSedyx) [inline todo_dj/urls.py]: use 1 space, pls

File: items/models.py (hunk 3 above)
Comment (ghost) [inline items/models.py]: :+1: it's not your model!

File: items/resources.py (hunk 4 above)
Comment (ghost) [inline items/resources.py]: spaces missed and you can avoid parenthesis here

File: items/resources.py (hunk 6 above)
Comment (ghost) [inline items/resources.py]: it's not work in this way. Just call `self.is_authenticated(request)` , it will generate Unathorized response if needed

**Pull request closed without merging**